nwb_conversion_tools/utils/si090nwbephyswriter.py

```diff
@@ -78,8 +78,16 @@ def add_to_nwb(self):
             self.add_sorting()
             self.add_waveforms()
 
-    def _get_traces(self, channel_ids=None, start_frame=None, end_frame=None, return_scaled=True):
-        return self.recording.get_traces(channel_ids=None, start_frame=None, end_frame=None, return_scaled=True).T
+    def get_num_segments(self):
+        return self.object_to_write.get_num_segments()
+
+    def _get_num_frames(self, segment_index=0):
+        if self.recording is not None:
+            return self.recording.get_num_frames(segment_index=segment_index)
+
+    def _get_traces(self, channel_ids=None, start_frame=None, end_frame=None, return_scaled=True, segment_index=0):
+        return self.recording.get_traces(channel_ids=None, start_frame=None, end_frame=None,
+                                         return_scaled=True, segment_index=segment_index).T
```

Comment:
@Saksham20 I think it's better to transpose the SI013 and avoid transposing [here](https://github.com/catalystneuro/nwb-conversion-tools/blob/ephys_writer/nwb_conversion_tools/utils/basenwbephyswriter.py#L579) to make it more efficient